templates/single-product/add-to-cart/variable.php

```diff
@@ -16,12 +16,13 @@
 
 if ( ! function_exists( 'print_attribute_radio' ) ) {
 	function print_attribute_radio( $checked_value, $value, $label, $name ) {
+		global $product;
 		// This handles < 2.4.0 bw compatibility where text attributes were not sanitized.
 		$checked = sanitize_title( $checked_value ) === $checked_value ? checked( $checked_value, sanitize_title( $value ), false ) : checked( $checked_value, $value, false );
 
 		$input_name = 'attribute_' . esc_attr( $name ) ;
 		$esc_value = esc_attr( $value );
-		$id = esc_attr( $name . '_v_' . $value );
+		$id = esc_attr( $name . '_v_' . $value . $product->id ); //added product ID at the end of the name to target single products
```

Comment:
we should use `->get_id()` instead of `->id` as this is deprecated in the latest woocommerce version